File: client/task-master/src/app/models/BoardAccessLevel.ts
import { UserAccessLevel } from './UserAccessLevel';
import { AccessLevelType } from './enum/AccessLevelType';

// Объект уровня доступа к доске
export class BoardAccessLevel {
  // Флаг публичного доступа к доске
  isPublic: boolean;
  // Уровни доступа пользователей к доске
  personalAccessLevels: UserAccessLevel[] = [];
  // Тип уровня доступа по умолчанию
  defaultAccessLevelType: AccessLevelType;

  // Метод для присвоения значений из частичного уровня доступа к доске
  assign(boardAccessLevel: Partial<BoardAccessLevel>) {
    this.isPublic = boardAccessLevel?.isPublic ?? this.isPublic;
    this.personalAccessLevels =
      boardAccessLevel?.personalAccessLevels ?? this.personalAccessLevels;
    this.defaultAccessLevelType =
      boardAccessLevel?.defaultAccessLevelType ?? this.defaultAccessLevelType;
  }
}

File: server/TaskMaster/TaskMaster.DataWebApi/Models/BoardAccessLevel.cs
using TaskMaster.DataAccessModule.Constants;
using TaskMaster.DataAccessModule.Models;

namespace TaskMaster.DataWebApi.Models
{
	/// <summary>
	/// Уровень доступа к доске.
	/// </summary>
	public class BoardAccessLevel
	{
		/// <summary>
		/// Указывает, является ли доска общедоступной.
		/// </summary>
		public bool IsPublic { get; set; }

		/// <summary>
		/// Список уровней доступа для отдельных пользователей.
		/// </summary>
		public List<UserAccessLevel> PersonalAccessLevels { get; set; }

		/// <summary>
		/// Тип уровня доступа по умолчанию.
		/// </summary>
		public AccessLevelType DefaultAccessLevelType { get; set; }
	}
}
